File: go/arrays_and_hashing/two_sum.go
package main

func twoSum(nums []int, target int) []int {
    m := make(map[int]int)

    for i, num := range nums {
        value, exists := m[num]
        if exists {
            return []int{i, value}
        }

        diff := target - num
        m[diff] = i
    }

    return []int{}
}

File: python/arrays_and_hashing/two_sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        diff_map = {}
        for idx, num in enumerate(nums):
            diff = target - num
            if diff in diff_map:
                return [idx, diff_map[diff]]

            diff_map[num] = idx

        return [-1, -1]
